### PR TITLE
Feature/metric benchmark

### PR DESCRIPTION
**Pull request by ambushwork**
### What does this PR do?

Merging metric benchmark feature branch to develop.

### Motivation

RUM-5553
RUM:5550

### Additional Notes

**All the commits have been reviewed and approved before**

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: tools/benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+4. Stop all gauges with meter:
+
+```kotlin
+meter.stopGauges()
+```
+
+## Observe Metrics
+
+All the metrics are uploaded
+through [Datadog submit metrics API](https://docs.datadoghq.com/api/latest/metrics/#submit-metrics).
```
Comment (buraizu) [inline tools/benchmark/README.md:43]:
```suggestion
through the [Datadog metric submission API](https://docs.datadoghq.com/api/latest/metrics/#submit-metrics).
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a minor, optional suggestion

File: tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.benchmark
+
+/**
+ * Describes the configuration to be used in Datadog open telemetry exporter.
+ * @param serviceName the name of the service.
+ * @param resource the name of resource.
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt:12]:
```suggestion
 * @param resource the name of the resource.
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+ * @param applicationName the name of host application.
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt:13]:
```suggestion
 * @param applicationName the name of the host application.
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+ * @param applicationVersion the version of host application.
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt:14]:
```suggestion
 * @param applicationVersion the version of the host application.
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+ * @param applicationId the id of the host application.
+ * @param apiKey the api key for submitting metrics to datadog end points.
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt:16]:
```suggestion
 * @param apiKey the api key for submitting metrics to datadog endpoints.
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+ * @param run the run description of benchmark test.
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt:17]:
```suggestion
 * @param run the run description of the benchmark test.
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+ * @param scenario the scenario of benchmark.
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt:18]:
```suggestion
 * @param scenario the scenario of the benchmark.
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+    val apiKey: String,
+    val run: String? = null,
+    val scenario: String? = null,
+    val endPoint: EndPoint = EndPoint.US1,
+    val intervalInSeconds: Long = DEFAULT_INTERVAL_IN_SECONDS
+) {
+    /**
+     * A Builder class for a [DatadogExporterConfiguration].
+     *
+     * @param apiKey api key to submit metrics to Datadog endpoint
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/DatadogExporterConfiguration.kt:37]:
```suggestion
     * @param apiKey api key to submit metrics to the Datadog endpoint
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/DatadogMeter.kt [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+            reader.stop()
+            gaugesByMetricName[reader.metricName()]?.close()
+            gaugesByMetricName[reader.metricName()] = NoOpObservableDoubleGauge()
+        }
+    }
+
+    companion object {
+
+        /**
+         * Creates an instance of [DatadogMeter] with given configuration.
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/DatadogMeter.kt:85]:
```suggestion
         * Creates an instance of [DatadogMeter] with the given configuration.
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/internal/MetricRequestBodyBuilder.kt [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+                mapOf(KEY_RESOURCES_NAME to it, KEY_RESOURCES_TYPE to "")
+            }
+
+        )
+    }
+
+    companion object {
+
+        private const val RESOURCE_ATTRIBUTE_NAME_KEY = "service.name"
+
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/internal/MetricRequestBodyBuilder.kt:101]:
```suggestion
```

File: tools/benchmark/src/main/java/com/datadog/benchmark/internal/RequestBodyBuilder.kt [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.benchmark.internal
+
+import com.google.gson.JsonElement
+
+internal interface RequestBodyBuilder<T> {
+
```
Comment (jonathanmos) [inline tools/benchmark/src/main/java/com/datadog/benchmark/internal/RequestBodyBuilder.kt:12]:
```suggestion
```

**Review by jonathanmos — APPROVED** [review]
on the whole LGTM - a few very minor comments

**Pull request merged**